### ISSUE TITLE
base_path is not optional

### ISSUE THREAD

**Issue by croemmich** [configcat/terraform-provider-configcat#42]
We just pulled down the `v2` branch to test out v2 configs. Upon running against our existing project we're seeing this error:

```
╷
│ Error: Missing base_path
│ 
│   with provider["registry.terraform.io/configcat/configcat"],
│   on providers.tf line 1, in provider "configcat":
│    1: provider "configcat" {
│ 
│ The provider cannot create the ConfigCat Public Management API client as there is a missing or empty value for the base_path.Set the base_path value in the configuration or use the CONFIGCAT_BASE_PATH
│ environment variable. If either is already set, ensure the value is not empty.
╵
``` 

It would appear that `base_path` is not actually optional as of the v4.0.0 release. We just noticed when updating from v3.1.0 to the v2 branch.

The provider works as expected when `base_path` is set in the provider block.

**Comment by laliconfigcat** [comment]
Hello @croemmich,

You are totally right, the basePath should fallback to api.configcat.com, thanks for reporting the issue.

**Comment by croemmich** [comment]
Thanks for the quick fix @laliconfigcat!

**Comment by laliconfigcat** [comment]
Hello @croemmich ,

I just released [v4.1.0](https://github.com/configcat/terraform-provider-configcat/releases/tag/v4.1.0) from the provider. I hope it will work for you.
Apart from this small hiccup, do you have any feedback for the V2 branch?

**Comment by croemmich** [comment]
@laliconfigcat, Nope, it appears to be working as expected! That said, we're not managing settings using Terraform, only the configs, so the only change to us was setting `evaluation_version = "v2"` on config resources.

I do have some general thoughts on the Terraform provider (probably more so the underlying public API.) We ended up needing to build a custom provider on top of the dashboard API for the following features:

- Get Existing Invitations
- Delete Invitations
- Update Product Preferences (need to set mandatory notes on all products for compliance reasons)
- Setting organization member permissions (admin and billing)
- CRUD webhooks
- Get webhook signing keys

That's less than ideal for a few reasons:
- More prone to breakage (since it's not a supported API)
- Much lower rate limit
- Requires us to have user/password auth enabled, which we're rather users not be able to use that method

Additionally, we intermittently encounter what I think is a race condition. We're using the `configcat_sdkkeys` data source, occasionally, we get a 404 when trying to get keys for a newly created config/environment. I haven't had a chance to test it thoroughly, but I wonder if order of making environments vs configs results in a different behavior and the key not being generated. If it go in to the ConfigCat UI and add a setting for the config/environment key returning a 404 it immediately starts working.

I've talked to Csilla about some of this, but I'll shoot you a more detailed Slack message when I get the time.

Thanks for the great product and support thus far!

**Comment by laliconfigcat** [comment]
Hey @croemmich ,

Awesome to hear that the v2 config creation is working as expected. 

You are totally right, all of the mentioned endpoints should exist on our Public Management API. Some of these were already on our backlog, and I created a ticket for the rest of the missing endpoints. We'll keep you updated if we made any progress with them.

Could you please show me some of your terraform configurations regarding the 404 issue? I'd be interested in the configcat_config, configcat_environment resources and the configcat_sdkkeys data source configuration.

**Comment by laliconfigcat** [comment]
Hello @croemmich ,

FYI: I merged the V2 pull request yesterday and released v5.0.0 from the provider so you don't have to rely on a locally built provider anymore. Let me know if you have any thoughts about it.

Cheers,

**Comment by laliconfigcat** [comment]
Hello @croemmich ,

I just wanted to let you know that we released a new version of the Terraform Provider that allows you to manage webhooks and product preferences.

I hope you'll like it.

Cheers
